recipes/LibriSpeech/SSL/train.py

```diff
@@ -0,0 +1,518 @@
+        else:
+            wav_lst.append(sig)
+            latent_length = get_out_len_fn(torch.as_tensor(sig.size(-1)))
+            latent_length_lst.append(latent_length.item())
+    bs = len(wav_lst)
+    wavs_padded, wav_lens = batch_pad_right(wav_lst)
+
+    batch_time_len = max(latent_length_lst)
+    mask = compute_mask(
+        (bs, batch_time_len,), latent_length_lst, mask_prob, mask_length
+    )
+    return (
+        torch.as_tensor(wavs_padded),
+        torch.as_tensor(wav_lens),
+        torch.as_tensor(mask, dtype=torch.bool),
+    )
+
+
+def _setup_wandb(hparams):
+    global wandb
+    import wandb
+
+    id_file = osp.join(hparams["output_folder"], "id")
+    if not hparams["distributed_launch"] or (
+        hparams["distributed_launch"] and int(os.environ["RANK"]) == 0
+    ):
+        # This should run when not using distributed training OR
+        # when using and this is the rank 0 node of distributed training.
+        dct = {
+            k: hparams.get(k)
+            for k in hparams.keys()
+            if hparams[k].__class__.__name__ in ("str", "int", "float", "bool")
+        }  # collects hparams to log
+        if os.path.exists(id_file):
+            id, name = open(id_file).read().split()
+            logger.info(
+                f"Loading ID {id} and name {name} for wandb from existing file {id_file}"
+            )
+            run = wandb.init(
+                project="wav2vec", resume="allow", config=dct, id=id, name=name,
+            )
+        else:
+            run = wandb.init(project="wav2vec", resume="allow", config=dct)
+            id = run.id
+            name = run.name
+            logger.info(
+                f"Creating new ID {id} and name {name} for wandb, putting in file {id_file}"
+            )
+            with open(id_file, "w") as fh:
+                fh.write(f"{id} {name}")
+
+
+def main():
+    logger.setLevel(logging.INFO)
+    print(sys.argv[1:])
+    hparams_file, run_opts, overrides = sb.parse_arguments(sys.argv[1:])
+
+    sb.utils.distributed.ddp_init_group(run_opts)
+
+    with open(hparams_file) as fin:
+        hparams = load_hyperpyyaml(fin, overrides)
+    hparams.update(run_opts)
+
+    sb.create_experiment_directory(
+        experiment_directory=hparams["output_folder"],
+        hyperparams_to_save=hparams_file,
+        overrides=overrides,
+    )
+
+    from librispeech_prepare import prepare_librispeech
+
+    run_on_main(
+        prepare_librispeech,
+        kwargs={
+            "data_folder": hparams["data_folder"],
+            "tr_splits": hparams["train_splits"],
+            "dev_splits": hparams["dev_splits"],
+            "te_splits": hparams["test_splits"],
+            "save_folder": hparams["output_folder"],
+            "merge_lst": hparams["train_splits"],
+            "merge_name": "train.csv",
+            "skip_prep": hparams["skip_prep"],
+        },
+    )
+
+    if hparams["use_wandb"]:
```

Comment:
Can we make wandb compliant with offline mode here ? I mean, 99% of large cluster don't have access to internet on the compute nodes

Reply:
what do you mean? if this is false then wandb wont be used (so that should be fine for offline mode)